### PR TITLE
Introduce Azure Traffic Manager 2018-04-01 API version with new features

### PR DESCRIPTION
**Pull request by ttlcgl**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/examples/Endpoint-DELETE-External.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "parameters": {
+    "subscriptionId": "e68d4145-c9ae-4667-925d-c51c8d88ad73",
+    "resourceGroupName": "azuresdkfornetautoresttrafficmanager1421",
+    "profileName": "azsmnet6386",
+    "endpointType": "ExternalEndpoints",
+    "endpointName": "azsmnet7187",
+    "api-version": "2017-05-01"
```
Comment (dsgouda) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/examples/Endpoint-DELETE-External.json]: I think we should verify whether the versions mentioned in these examples match with the version number in the directory path here.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/examples/HeatMap-GET-With-Null-Values.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+  "parameters": {
+    "subscriptionId": "e68d4145-c9ae-4667-925d-c51c8d88ad73",
```
Comment (dsgouda) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/examples/HeatMap-GET-With-Null-Values.json]: nit: we recommend not exposing production style subscriptionids

**Review by dsgouda — COMMENTED** [review]
Partial review since @sergey-shandar will be taking over this one

**Comment by tatlicioglu** [comment]
@sergey-shandar 
Thanks Sergey for remarks. I've replaced explicit subscription ID with placeholder and I've fixed all API versions in examples for stable\2018-04-01 version. I've added two separate commits for each remark for easier review.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 3]
```diff
@@ -0,0 +1,1578 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "TrafficManagerManagementClient",
+    "version": "2018-03-01"
```
Comment (ravbhatnagar) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: folder name is 2018-04-01 but the version here is still 2018-03-01. Do you intend to add the APIs in a newer APi-version or existing?

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 4]
```diff
@@ -0,0 +1,1578 @@
...
+            }
+          }
+        },
+        "x-ms-examples": {
+          "TrafficManagerUserMetricsKeys-GET": {
+            "$ref": "./examples/TrafficManagerUserMetricsKeys-GET.json"
+          }
+        }
+      },
+      "put": {
```
Comment (ravbhatnagar) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: PUT on a collection is not allowed/supported by ARM. Please include the instancename in the URL. 
It should be /subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/{metricKeyName}. If it does not really have a name, you can put a default or current as the segment. The pattern wont work for ARM in the current shape. Check other resource types in this spec for reference.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 5]
```diff
@@ -0,0 +1,1578 @@
...
+            }
+          }
+        },
+        "x-ms-examples": {
+          "TrafficManagerUserMetricsKeys-PUT": {
+            "$ref": "./examples/TrafficManagerUserMetricsKeys-PUT.json"
+          }
+        }
+      },
+      "delete": {
```
Comment (ravbhatnagar) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: same comment as above.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 6]
```diff
@@ -0,0 +1,1578 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Class representing the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "UserMetricsProperties": {
+      "properties": {
+        "key": {
```
Comment (ravbhatnagar) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: what does this key indicate and how is it used? is it considered a secret? Is it readOnly that the service generates for the user or something which needs to be provided by the user.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 7]
```diff
@@ -0,0 +1,1578 @@
...
+          "HeatMap-GET-With-TopLeft-BotRight": {
+            "$ref": "./examples/HeatMap-GET-With-TopLeft-BotRight.json"
+          },
+          "HeatMap-GET-With-Null-Values": {
+            "$ref": "./examples/HeatMap-GET-With-Null-Values.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys": {
```
Comment (ravbhatnagar) [inline specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: Is it at subscription scope or should it live at resource group scope?

**Comment by ravbhatnagar** [comment]
@tatlicioglu - please take a look at the comments and feel free to set something up if there are questions.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json (hunk 3 above)
Comment (allencal) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: That was in error. Fixed in https://github.com/Azure/azure-rest-api-specs/pull/3360

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json (hunk 7 above)
Comment (allencal) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: It is at subscription scope.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json (hunk 6 above)
Comment (allencal) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: It is not a secret. Is it essentially a unique ID that is shared knowledge between the subscription and Microsoft. It is read-only. They "PUT" to create (or refresh) a key, and then we create it for them.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json (hunk 4 above)
Comment (allencal) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: Should be /subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json (hunk 5 above)
Comment (allencal) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: Same

**Comment by allencal** [comment]
I have created a new PR https://github.com/Azure/azure-rest-api-specs/pull/3361 since I did not have auth to update Mehmet's PR. The two PR's cross-reference one another now. All above comments have been addressed in 3361.

File: specification/trafficmanager/resource-manager/readme.ruby.md [hunk 8]
```diff
@@ -17,6 +17,9 @@ batch:
   - tag: package-2017-03
   - tag: package-2017-05
   - tag: package-2017-09-preview-only
+  - tag: package-2018-02
+  - tag: package-2018-03
```
[the remark this answers is not in the document]
Comment (allencal) [reply specification/trafficmanager/resource-manager/readme.ruby.md]: I don't think this is supposed to be part of the PR. Omitted them from the new PR (https://github.com/Azure/azure-rest-api-specs/pull/3361)

[the remark this answers is not in the document]
Comment (allencal) [reply specification/trafficmanager/resource-manager/readme.ruby.md]: Okay I figured this out. readme.md was missed from this PR. Adding in 3361.

**Comment by sergey-shandar** [comment]
See @3361

**Pull request closed without merging**